astropy/utils/data.py

```diff
@@ -977,145 +1000,148 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None):
+                                    .format(size, bytes_read),
+                                    content=None,
                                 )
-                    if size is not None and bytes_read < size:
-                        raise urllib.error.ContentTooShortError(
-                            "File was supposed to be {} bytes but we only "
-                            "got {} bytes. Download failed."
-                            .format(size, bytes_read),
-                            content=None,
-                        )
-                except BaseException:
-                    if os.path.exists(f.name):
-                        os.remove(f.name)
-                    raise
-
-        if cache:
-            _acquire_download_cache_lock()
-            try:
-                with shelve.open(urlmapfn) as url2hash:
-                    # We check now to see if another process has
-                    # inadvertently written the file underneath us
-                    # already
-                    if url_key in url2hash:
-                        return url2hash[url_key]
-                    local_path = os.path.join(dldir, hash.hexdigest())
-                    shutil.move(f.name, local_path)
-                    url2hash[url_key] = local_path
-            finally:
-                _release_download_cache_lock()
-        else:
-            local_path = f.name
-            if missing_cache:
-                msg = ('File downloaded to temporary location due to problem '
-                       'with cache directory and will not be cached.')
-                warn(CacheMissingWarning(msg, local_path))
-            if conf.delete_temporary_downloads_at_exit:
-                global _tempfilestodel
-                _tempfilestodel.append(local_path)
-    except urllib.error.URLError as e:
-        if hasattr(e, 'reason') and hasattr(e.reason, 'errno') and e.reason.errno == 8:
-            e.reason.strerror = e.reason.strerror + '. requested URL: ' + remote_url
-            e.reason.args = (e.reason.errno, e.reason.strerror)
-        raise e
-    except socket.timeout as e:
-        # this isn't supposed to happen, but occasionally a socket.timeout gets
-        # through.  It's supposed to be caught in `urrlib2` and raised in this
-        # way, but for some reason in mysterious circumstances it doesn't. So
-        # we'll just re-raise it here instead
-        raise urllib.error.URLError(e)
+                        except BaseException:
+                            if os.path.exists(f.name):
+                                try:
+                                    os.remove(f.name)
+                                except OSError:
+                                    pass
+                            raise
+
+                # Success!
+                break
+
+        except urllib.error.URLError as e:
+            if hasattr(e, 'reason') and hasattr(e.reason, 'errno') and e.reason.errno == 8:
```

Comment:
For ease of future maintenance, please explain what 8 means.

Reply:
Good question. This exception may have a name that allows us to leave out errno entirely. I'll check.